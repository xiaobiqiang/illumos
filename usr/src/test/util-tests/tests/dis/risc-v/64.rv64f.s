/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test our disassembly of the RV64F instructions. Instructions are ordered per
 * the ISA manual.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	fcvt.l.s	t0, ft1
	fcvt.l.s	t1, ft2, rne
	fcvt.l.s	t2, ft3, rtz
	fcvt.l.s	t3, ft4, rdn
	fcvt.l.s	t4, ft5, rup
	fcvt.l.s	t5, ft6, rmm

	fcvt.lu.s	t1, ft0
	fcvt.lu.s	t2, ft1, rne
	fcvt.lu.s	t3, ft2, rtz
	fcvt.lu.s	t4, ft3, rdn
	fcvt.lu.s	t5, ft4, rup
	fcvt.lu.s	t6, ft5, rmm

	fcvt.s.l	ft1, t0
	fcvt.s.l	ft2, t1, rne
	fcvt.s.l	ft3, t2, rtz
	fcvt.s.l	ft4, t3, rdn
	fcvt.s.l	ft5, t4, rup
	fcvt.s.l	ft6, t5, rmm

	fcvt.s.lu	ft1, t0
	fcvt.s.lu	ft2, t1, rne
	fcvt.s.lu	ft3, t2, rtz
	fcvt.s.lu	ft4, t3, rdn
	fcvt.s.lu	ft5, t4, rup
	fcvt.s.lu	ft6, t5, rmm
.size libdis_test, [.-libdis_test]
