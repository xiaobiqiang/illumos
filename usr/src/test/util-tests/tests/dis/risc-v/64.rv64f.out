    libdis_test:      d3 f2 20 c0        fcvt.l.s t0,ft1
    libdis_test+0x4:  53 03 21 c0        fcvt.l.s t1,ft2,rne
    libdis_test+0x8:  d3 93 21 c0        fcvt.l.s t2,ft3,rtz
    libdis_test+0xc:  53 2e 22 c0        fcvt.l.s t3,ft4,rdn
    libdis_test+0x10: d3 be 22 c0        fcvt.l.s t4,ft5,rup
    libdis_test+0x14: 53 4f 23 c0        fcvt.l.s t5,ft6,rmm
    libdis_test+0x18: 53 73 30 c0        fcvt.lu.s t1,ft0
    libdis_test+0x1c: d3 83 30 c0        fcvt.lu.s t2,ft1,rne
    libdis_test+0x20: 53 1e 31 c0        fcvt.lu.s t3,ft2,rtz
    libdis_test+0x24: d3 ae 31 c0        fcvt.lu.s t4,ft3,rdn
    libdis_test+0x28: 53 3f 32 c0        fcvt.lu.s t5,ft4,rup
    libdis_test+0x2c: d3 cf 32 c0        fcvt.lu.s t6,ft5,rmm
    libdis_test+0x30: d3 f0 22 d0        fcvt.s.l ft1,t0
    libdis_test+0x34: 53 01 23 d0        fcvt.s.l ft2,t1,rne
    libdis_test+0x38: d3 91 23 d0        fcvt.s.l ft3,t2,rtz
    libdis_test+0x3c: 53 22 2e d0        fcvt.s.l ft4,t3,rdn
    libdis_test+0x40: d3 b2 2e d0        fcvt.s.l ft5,t4,rup
    libdis_test+0x44: 53 43 2f d0        fcvt.s.l ft6,t5,rmm
    libdis_test+0x48: d3 f0 32 d0        fcvt.s.lu ft1,t0
    libdis_test+0x4c: 53 01 33 d0        fcvt.s.lu ft2,t1,rne
    libdis_test+0x50: d3 91 33 d0        fcvt.s.lu ft3,t2,rtz
    libdis_test+0x54: 53 22 3e d0        fcvt.s.lu ft4,t3,rdn
    libdis_test+0x58: d3 b2 3e d0        fcvt.s.lu ft5,t4,rup
    libdis_test+0x5c: 53 43 3f d0        fcvt.s.lu ft6,t5,rmm
